Update documentation for automatic model undeployment (#7083)
opensearch-project/documentation-website: _ml-commons-plugin/api/model-apis/undeploy-model.md

  }
}
```
### Automatically undeploy a model based on TTL

> Starting with OpenSearch version 2.14, models can be automatically undeployed from memory based on the pre-defined TTL (time-to-live) since they were last accessed or used. To define a TTL that auto undeploys a model, include the below ModelDeploySetting in your ML mode.

Reviewer: ```suggestion
Starting with OpenSearch  2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL), when the model was last accessed or used.. To define a TTL that auto undeploys a model, include the following `ModelDeploySetting` in your ML mode.
```

```json
PUT /_plugins/_ml/models/COj7K48BZzNMh1sWedLK
{

>     "deploy_setting": {"model_ttl_minutes" : 100}

Reviewer 2: User should update model TTL when model undeployed.

  }
}
```
### Automatically undeploy a model based on TTL

> Starting with OpenSearch version 2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL) when the model was last accessed or used. To define a TTL that auto undeploys a model, include the following ModelDeploySetting in your ML mode. Note that model TTLs are checked periodically by a syn_up cron job, so the maximum time a model lives in the memory could be TTL + sync_up_job_interval. The default cron job interval is 10 seconds. To update the cron job internal, use the following cluster setting.

Reviewer: ```suggestion
Starting with OpenSearch  2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL) when the model was last accessed or used. To define a TTL that auto undeploys a model, include the following `ModelDeploySetting` in your ML node. Note that model TTLs are checked periodically by a `syn_up` cron job, so the maximum time a model lives in the memory could be TTL + the `sync_up_job_` interval. The default cron job interval is 10 seconds. To update the cron job internally, use the following cluster setting:
```

> Starting with OpenSearch  2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL) when the model was last accessed or used. To define a TTL that auto undeploys a model, include the following `ModelDeploySetting` in your ML node. Note that model TTLs are checked periodically by a `syn_up` cron job, so the maximum time a model lives in the memory could be TTL + the `sync_up_job_` interval. The default cron job interval is 10 seconds. To update the cron job internally, use the following cluster setting:

Reviewer 3: ```suggestion
Starting with OpenSearch 2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL) when the model was last accessed or used. To define a TTL that automatically undeploys a model, include the following `ModelDeploySetting` in your machine learning (ML) node. Note that model TTLs are checked periodically by a `syn_up` cron job, so the maximum time that a model lives in memory could be TTL + the `sync_up_job_` interval. The default cron job interval is 10 seconds. To update the cron job internally, use the following cluster setting:
```

> Starting with OpenSearch version 2.14, models can be automatically undeployed from memory based on the predefined time-to-live (TTL) when the model was last accessed or used. To define a TTL that auto undeploys a model, include the following ModelDeploySetting in your ML mode. Please note that model TTLs are checked periodically by a syn_up cron job, so the maximum time a model lives in the memory could be TTL + sync_up_job_interval. The default cron job interval is 10 seconds. To update the cron job internal, use the following cluster setting.

Reviewer 2: Check the `GitHub Actions / style-job` 
> Using 'Please' is unnecessary. Remove

Reviewer 2: If I'm understanding correctly, the user can indeed update the model's Time To Live (TTL) after it has been deployed without encountering an exception. However, this TTL update will not take effect immediately. A deployment or redeployment of the model is required for the updated TTL to come into effect.

```json
PUT /_cluster/settings
{
    "persistent": {
        "plugins.ml_commons.sync_up_job_interval_in_seconds": 10
    }
}
```

> #### Example request: Creating a model with TTL

Reviewer 3: ```suggestion
#### Example request: Creating a model with a TTL
```

```json
POST /_plugins/_ml/models/_register
 {
   "name": "Sample Model Name",
   "function_name": "remote",
   "description": "test model",
   "connector_id": "-g1nOo8BOaAC5MIJ3_4R",

>    "deploy_setting": {"model_ttl_minutes": 100}

Reviewer 2: Can user use this for local model?

Author: Yes it's for all models. Only local model require manual deploy. Everything else is automated in terms of model deploy/un-deploy.

 }
```

> #### Example request: Updating a model with TTL when the model is undeployed

Reviewer 3: ```suggestion
#### Example request: Updating a model with a TTL when the model is undeployed
```
